lib/internal/streams/end-of-stream.js

```diff
@@ -67,6 +67,24 @@ function eos(stream, opts, callback) {
     }
   };
 
+  const readableDestroyed = stream._readableState &&
+    stream._readableState.destroyed;
+  const writableDestroyed = stream._writableState &&
+    stream._writableState.destroyed;
+
+  if (readableDestroyed || writableDestroyed) {
+    let err;
+    if (readable) {
+      if (!stream._readableState || !stream._readableState.endEmitted)
+        err = new ERR_STREAM_PREMATURE_CLOSE();
+    }
+    if (writable) {
+      if (!stream._writableState || !stream._writableState.finished)
+        err = new ERR_STREAM_PREMATURE_CLOSE();
+    }
+    process.nextTick(callback.bind(stream, err));
```

Comment:
should we return after this nextTick?

Reply:
Maybe, I think the reason I skipped the return is in case an error is emitted after `destroyed` is set. But I don't think that is actually a relevant case.